dkim/header.go

```diff
@@ -92,37 +92,60 @@ func parseHeaderParams(s string) (map[string]string, error) {
 	return params, nil
 }
 
-func formatHeaderParams(params map[string]string) string {
-	keys := make([]string, 0, len(params))
-	found := false
-	for k := range params {
-		if k == "b" {
-			found = true
-		} else {
-			keys = append(keys, k)
-		}
-	}
-	sort.Strings(keys)
-	if found {
-		keys = append(keys, "b")
-	}
+func formatHeaderParams(headerFieldName string, params map[string]string) string {
+	keys, bvalue := sortParams(params)
 
 	var s string
+	var line string
 	first := true
+
 	for _, k := range keys {
 		v := params[k]
 
 		if first {
 			first = false
+			line += fmt.Sprintf("%v: %v=%v;", headerFieldName, k, v)
+			continue
 		} else {
-			s += " "
+			nextLength := 3 + len(line) + len(v) + len(k)
+			if nextLength > 75 {
+				s += line + crlf
+				line = ""
+			}
+			line = fmt.Sprintf("%v %v=%v;", line, k, v)
 		}
+	}
 
-		s += k + "=" + v + ";"
+	if line != "" {
+		s += line
 	}
+
+	if bvalue != "" {
```

Comment:
Empty string is a valid value. The DKIM-Signature contents are signed with `b=`.

Reply:
Oh thanks, I've got it now! It seems it works :D